### ISSUE TITLE
Timeout broken in elasticesearch 2.0.0

### ISSUE THREAD

**Issue by djzort** [harisekhon/nagios-plugins#74]
the timeout= parameter now needs a unit, however its not applied consistently. At a minimum the _cluster/health call now does. See also https://www.elastic.co/guide/en/elasticsearch/reference/2.0/cluster-health.html

BAD
curl -sS -XGET 'localhost/_cluster/health?pretty=true&timeout=12'
{
  "error" : {
    "root_cause" : [ {
      "type" : "parse_exception",
      "reason" : "Failed to parse setting [timeout] with value [12] as a time value: unit is missing or unrecognized"
    } ],
    "type" : "parse_exception",
    "reason" : "Failed to parse setting [timeout] with value [12] as a time value: unit is missing or unrecognized"
  },
  "status" : 400
}

GOOD
curl -sS -XGET 'localhost/_cluster/health?pretty=true&timeout=12s'
{
  "cluster_name" : "elasticsearch200-prod",
  "status" : "green",
  "timed_out" : false,
  "number_of_nodes" : 7,
  "number_of_data_nodes" : 5,
  "active_primary_shards" : 135,
  "active_shards" : 270,
  "relocating_shards" : 0,
  "initializing_shards" : 0,
  "unassigned_shards" : 0,
  "delayed_unassigned_shards" : 0,
  "number_of_pending_tasks" : 0,
  "number_of_in_flight_fetch" : 0,
  "task_max_waiting_in_queue_millis" : 0,
  "active_shards_percent_as_number" : 100.0
}

The generation of the timeout=X parameter is deep in your libraries, i would like to send a PR but i dont want to mess around inside something personal that youre using so widely. 

It does seem though, that elasticsearch 1.4 also supports (requires?) a unit. As per https://www.elastic.co/guide/en/elasticsearch/reference/1.4/cluster-health.html

Also 0.9 https://www.elastic.co/guide/en/elasticsearch/reference/0.90/cluster-health.html

**Comment by djzort** [comment]
This seems to work for me on 2.0.0

diff --git a/HariSekhon/Elasticsearch.pm b/HariSekhon/Elasticsearch.pm
index 6c54d23..67f726b 100644
--- a/HariSekhon/Elasticsearch.pm
+++ b/HariSekhon/Elasticsearch.pm
@@ -139,7 +139,7 @@ sub curl_elasticsearch_raw($;$$){
     } else {
         $url .= "?";
     }
-    $url .= "timeout=" . minimum_value($timeout - 1, 1);
+    $url .= "timeout=" . minimum_value($timeout - 1, 1) . 's';
     $url .= "&pretty=true" if $verbose >= 3 or $debug;
     #my $content = curl "http://$host:$port/$url", "Elasticsearch", undef, undef, \&elasticsearch_err_handler, $type;
     my $content = curl "http://$host:$port/$url", "Elasticsearch", undef, undef, undef, $type, $body;

**Comment by harisekhon** [comment]
@djzort

Thanks for raising this. I've made this change as all the elasticsearch code is tested in CI against a real Elasticsearch 1.x instance and would alert if this breaks anything. The test runs for all the depending plugins in this project have all passed, so I'm going to close this ticket as it's now fixed in master.

ps. you'll find sprintf littered here and there it's mainly used for security around format strings or if there is more/complicated interpolation but yes I don't always bother if it's simple 1 + 1 concats as it's needlessly more typing. To make you happy I've used sprintf since it looks better in this case :)
